### PR TITLE
UT of Charge::sum_band()

### PR DESCRIPTION
**Pull request by hongriTianqi**

### CODE REVIEW

File: source/src_ri/exx_lcao.h [hunk 1]
```diff
@@ -55,6 +55,7 @@ class Exx_Lcao
 private:
 	
 	int kmesh_times = 4;				// Peize Lin test
+#ifdef __MPI
```
Comment (caic99) [inline source/src_ri/exx_lcao.h:58]: This change breaks the compilation. Maybe @dyzheng can help.

File: source/src_pw/test/CMakeLists.txt [hunk 2]
```diff
@@ -14,3 +14,16 @@ AddTest(
...
+  LIBS ${math_libs} base cell symmetry md neighbor orb io ions lcao parallel mrrr pdiag pw ri driver xc ELPA::ELPA esolver
+  SOURCES sum_band_test.cpp 
+)
+
+install(FILES STRU DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+install(FILES H.pz-vbc.UPF DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+install(FILES KPT DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+install(FILES SPIN1_CHG DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+install(FILES WAVEFUNC1.txt DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+install(FILES BANDS_1.dat DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
```
Comment (caic99) [inline source/src_pw/test/CMakeLists.txt:29]: What does file name means, e.g. `data-H`, `WAVEFUNC1.txt`, `BANDS_1.dat`? You can add some more meaningful prefix/suffix to them.

File: source/src_pw/test/CMakeLists.txt [hunk 3]
```diff
@@ -14,3 +14,16 @@ AddTest(
 )
 
 install(FILES data-H DESTINATION ${CMAKE_CURRENT_BINARY_DIR})
+
+AddTest(
+  TARGET hsolver_sum_band
+  LIBS ${math_libs} base cell symmetry md neighbor orb io ions lcao parallel mrrr pdiag pw ri driver xc ELPA::ELPA esolver
```
Comment (caic99) [inline source/src_pw/test/CMakeLists.txt:20]: Do we need to link all libs?

**Review by caic99 — CHANGES_REQUESTED** [review]

**Comment by hongriTianqi** [comment]
This UT is too involved due to the use of many global class objects. A briefer one is expected to provide.

**Pull request closed without merging**